src/stringify.ts

```diff
@@ -26,159 +26,195 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-let BARE_KEY = /^[a-z0-9-_]+$/i
+let BARE_KEY = /^[a-z0-9-_]+$/i;
 
-type ExtendedType = ReturnType<typeof extendedTypeOf>
-function extendedTypeOf (obj: any) {
-	let type = typeof obj
-	if (type === 'object') {
-		if (Array.isArray(obj)) return 'array'
-		if (obj instanceof Date) return 'date'
-	}
+type ExtendedType = ReturnType<typeof extendedTypeOf>;
+function extendedTypeOf(obj: any) {
+  let type = typeof obj;
+  if (type === 'object') {
+    if (Array.isArray(obj)) return 'array';
+    if (obj instanceof Date) return 'date';
+  }
 
-	return type
+  return type;
 }
 
-function isArrayOfTables (obj: any[]) {
-	for (let i = 0; i < obj.length; i++) {
-		if (extendedTypeOf(obj[i]) !== 'object') return false
-	}
+function isArrayOfTables(obj: any[]) {
+  for (let i = 0; i < obj.length; i++) {
+    if (extendedTypeOf(obj[i]) !== 'object') return false;
+  }
 
-	return obj.length != 0
+  return obj.length != 0;
 }
 
-function formatString (s: string) {
-	return JSON.stringify(s).replace(/\x7f/g, '\\u007f')
+function formatString(s: string) {
+  // Use multi-line literal strings for strings containing newlines or long strings
+  if ((s.includes('\n') || s.length > 120) && !s.includes("'''")) {
+    // For long strings without newlines, try to break them at spaces
+    if (!s.includes('\n') && s.length > 120) {
+      let result = "'''\n";
+      let lineLength = 0;
+      const words = s.split(/(\s+)/);
+
+      for (let i = 0; i < words.length; i++) {
+        const word = words[i];
+        if (word === undefined) {
+          continue;
+        }
+        // If adding this word would exceed line length and we're not at the start of a line
+        if (lineLength > 0 && lineLength + word.length > 80) {
+          result += '\n';
+          lineLength = 0;
+        }
+
+        result += word;
+        lineLength += word.length;
```

Comment:
The words are appended without any spacing..? The problem here is that the split string has not been preserved and could've been any combination of whitespace characters, which makes it not feasible to reconstruct a byte-equivalent string from the gathered parts.